### DISCUSSION TITLE
Allow servers from all instances to be connected to with the app

### DISCUSSION THREAD

**Discussion by Cknight70** [stoatchat/discussions#164] [Feature Suggestions]
The Revolt desktop/mobile app should support joining servers from all different instances including sign ins from those other instances. Not to be confused with Federation.

**Benefits to revolt from implementing something like this:**
1. More cohesive experience for anyone who wants to join rooms from a self hosted instance
2. Incentive for self hosted instances to continue to update their instances and not split off from the main project
3. Decreases risk of censorship 
4. Decreases the chance of this project dieing if the main instance ever goes down.

Here's some quick examples of what doing something like this might look like:
![image](https://user-images.githubusercontent.com/28763729/134721354-7f12c3db-457b-4b83-9f6c-6b6cbb7bcf37.png)

Here's what some similar chat apps do:
Rocket.chat is not federated, but you can use multiple "servers" with the same app
![image](https://user-images.githubusercontent.com/28763729/134721530-520690e5-ca66-4a33-8883-742696fc2ceb.png)
![image](https://user-images.githubusercontent.com/28763729/134721608-f139cbff-860d-42cd-bd9d-79e1f26ce67c.png)

Element has an easy way to add new servers so you can browse their rooms 
![image](https://user-images.githubusercontent.com/28763729/134721891-e8859102-d299-42c2-82a6-6c9c977aa4d0.png)

**Comment by CaraAra** [comment]
This should certainly be a focus right after administration tools and voice chat. Self-hosting is one of the most awesome Revolt features and should be made as accessible to users as can be.
